Document regexes for showing only attributes or datasets in new ADIOS2 schema (#1068)
openPMD/openPMD-api: docs/source/backends/adios2.rst

The ADIOS2 backend will automatically recognize the layout that has been used by a writer when reading a dataset.

.. tip::

>    This schema does not use ADIOS2 attributes any more, making the ``bpls -a`` and ``bpls -A`` switches useless for openPMD datasets.

Reviewer: I think this needs an empty newline to render properly.

Maybe also a suggestion in the meaning:
```suggestion

   This schema does not use ADIOS2 attributes anymore, thus ``bpls -a`` and ``bpls -A`` attribute switches do not show openPMD attributes.
```

   Their functionality can be emulated via regexes:

>    * Print datasets and attributes: Default behavior

Reviewer: Is this what we mean here?
```suggestion
   * Print datasets and attributes: Default behavior (``bpls -a -l``)
```

Author: The only ADIOS2 attributes that we still use are internal ones such as `__openPMD_internal/openPMD2_adios2_schema`. So, the `-a` flag has (nearly) no influence. By default behavior, I mean "what happens if I call `bpls` without further parameters).

Reviewer: Ah, so essentially
```suggestion
   * Print datasets and attributes: No extra flags needed
```

>    * Print datasets only: ``bpls -e '.*/__data__$'``
>    * Print attributes only: ``bpls -e '^(.(?!/__data__$))*$'``

Reviewer: Maybe suggest the user `-l` for long here?
```suggestion
   * Print datasets only: ``bpls -l -e '.*/__data__$'``
   * Print attributes only: ``bpls -l -e '^(.(?!/__data__$))*$'``
```

Author: This listing is supposed to show how to select attributes or datasets from a BP file. The `-l` switch doesn't really have anything to do with that.
We can add a little sentence below that the `-l` flag is useful to show the values of scalars?

Reviewer: Oh I just add it by muscle memory because I always find it useful, that's why I was wondering if we just doc it that way.

The ADIOS2 backend will automatically recognize the layout that has been used by a writer when reading a dataset.

.. tip::

   This schema does not use ADIOS2 attributes anymore, thus ``bpls -a`` and ``bpls -A`` attribute switches do not show openPMD attributes.

>    Their functionality can be emulated via regexes:

Reviewer: ```suggestion
   Their functionality can be emulated in ``bpls`` via regexes:
```
